### PR TITLE
Landing position updates for precision landing.

### PR DESCRIPTION
**Pull request by kapacheuski**
## Problem

Precision landing works incorrectly - Target position is not updated during the descent above target.

## Solution
`_prepareLandSetpoints` needs to update `_land_position` continuously.

## Test coverage
Tests were performed on quadcopters with and without a fix.

### Log without fix
WithoutFix

### Log with fix
WithFix

### CODE REVIEW

File: src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp [hunk 1]
```diff
@@ -245,12 +245,12 @@ void FlightTaskAuto::_prepareLandSetpoints()
...
 	if (_type_previous != WaypointType::land) {
-		// initialize xy-position and yaw to waypoint such that home is reached exactly without user input
-		_land_position = Vector3f(_target(0), _target(1), NAN);
+		// initialize yaw
 		_land_heading = _yaw_setpoint;
 		_stick_acceleration_xy.resetPosition(Vector2f(_target(0), _target(1)));
 	}
-
+	// Update xy-position in case of landing position changes (etc. precision landing)
+	_land_position = Vector3f(_target(0), _target(1), NAN);
```
Comment (potaito) [inline src/modules/flight_mode_manager/tasks/Auto/FlightTaskAuto.cpp:253]: I wonder whether this affects land modes other than precision landing in any way.

But yes, I agree, for precision landing the setpoint needs to be updated continuously and not just during initialization.

**Review by potaito — APPROVED** [review]
I have two related PRs actually:
- Make precision landing a flight task, which would also fix the same issue: https://github.com/PX4/PX4-Autopilot/pull/18973
- Add yaw tracking to precision landing: https://github.com/PX4/PX4-Autopilot/pull/20487

**Comment by potaito** [comment]
@tstastny good to merge in my opinion
@kapacheuski thanks for your work :+1:

**Comment by dagar** [comment]
Merged in https://github.com/PX4/PX4-Autopilot/pull/20951.

**Pull request closed without merging**